lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -339,11 +339,8 @@ def __post_init__(self, *args: Any, **kwargs: Any) -> None:
         if self.service_principal_client_id or self.service_principal_key:
             add_secret(self.service_principal_client_id, mask=PATTERN_GUID)
             add_secret(self.service_principal_key)
-            if not self.service_principal_client_id or not self.service_principal_key:
-                raise LisaException(
-                    "service_principal_client_id and service_principal_key "
-                    "should be specified either both or not."
-                )
+            if not self.service_principal_client_id:
```

Comment:
It needs to split checks for client id and key. They don't need to be in the same if block.

Reply:
With the new logic, the exception is not needed. Because L341 checked it exists already.